core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -182,6 +184,8 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
   private var protocol: Option[String] = None
 
   private val members = new mutable.HashMap[String, MemberMetadata]
+  private var awaitingJoinCallbackMembers = 0
+  private val supportedProtocols = new mutable.HashMap[String, Integer]().withDefaultValue(0)
```

Comment:
Probably not too big of a deal, but should we clean this up when the count drops to 0 for a protocol?

Reply:
I don't think that's really needed. How many protocols will Kafka eventually have? 5? 10?
Even if there are 1000 protocols, the memory usage of 1000 hash table entries is still trivial